### PR TITLE
[CAMP-2446] Add Modal Component

### PR DESCRIPTION
**Pull request by Rajmohanrk**
**The following must be included in every PR. Please check all items before submitting your request.
If an item can not be satisfied, please provide a brief explanation as to why the item is not included.**

- [x] Rebased from Master prior to creation of pull request
- [x] Minimum 1 unit test that covers the component
- [x] All tests run and passed prior to creation of pull request
- [x] Component added to bit.dev server
- [x] New feature documentation added to bit.dev
- [x] Ran `yarn sync` to get most recent version of all components from bit.dev server
- [x] Sandbox example of your component added/updated to App.js

**Provide a brief description of your update:**
Add Modal Component 
Screenshot 2020-03-26 at 4 41 22 PM
Screenshot 2020-03-26 at 5 19 05 PM
Screenshot 2020-03-26 at 5 33 43 PM

### CODE REVIEW

File: src/examples/molecules/Modal/Example.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    <Card style={{ padding: '30px' }}>
+      <Button solid primary onClick={toggleModal}>
+        Open Modal
+      </Button>
+      <Modal
+        showModal={show}
+        modalTitle="modal title"
+        modalContent="Lorem ipsum dolor sit amet."
+        onModalClose={toggleModal}
+        className="modal--relative"
```
Comment (PRASHANTK22111993) [inline src/examples/molecules/Modal/Example.js]: do we really need this??
even if we need pass the prop and assign the class in components
```
<Modal relative />
<Modal fixed />
```

Comment (Rajmohanrk) [reply src/examples/molecules/Modal/Example.js]: we don't need that @PRASHANTK22111993 I have removed it.

Comment (PRASHANTK22111993) [reply src/examples/molecules/Modal/Example.js]: @Rajmohanrk do we still need in scss file .

```
  &--relative {
    position: absolute;
  }
```

Comment (Rajmohanrk) [reply src/examples/molecules/Modal/Example.js]: That would not be required as well

**Pull request merged**